### DISCUSSION TITLE
Communicating with Collections

### DISCUSSION THREAD

**Discussion by ThePalad1n** [pocketbase/pocketbase#130] [Q&A]
How would I be able to implement MVC using pocketbase as a framework. It seems that the SQLite connection rests with the app and using echo as a router, I cannot find a way to interface handlers with the collections.

**Comment by ganigeorgiev** [comment]
There are several ways to interact with the collections and their records.

If you just want to intercept the API requests, you can make use of the list, view, create, update and delete  hooks (there are examples once expanded) - https://pocketbase.io/docs/event-hooks/#record-hooks

Currently, this is not documented very well outside of the source code, but if you want to programmatically fetch, update, create records, collections, users, etc, you can also use the `app.Dao()` instance or the related form models to interact directly with the database (see example for creating a collection programmatically in https://github.com/pocketbase/pocketbase/discussions/125).

Exploring the source code could also help you. Here are for example the records related Dao methods - https://github.com/pocketbase/pocketbase/blob/master/daos/record.go

More advanced usage documentation is in the [roadmap](https://github.com/orgs/pocketbase/projects/2).

_Although I fully understand that godoc/source code documentation is not enough, keep in mind that PocketBase was released just last week and there is still a lot of space for improvements._

**Reply by ThePalad1n** [reply]
My team and I appreciate you taking the time to respond so quickly. We understand you're very new and will follow up as we have any more questions. Thanks for your help.

*Once we get this app up we would be more than happy to share it with you in support of this project*
Thanks again - IIQ
